src/main/java/io/pravega/perf/PerfStats.java

```diff
@@ -202,12 +220,133 @@ private long windowTimeMS(long time) {
         }
     }
 
+    /***
+     * Perf Recorder that records the number of bytes every second.  Percentiles are reported in human form, i.e. MB/s
+     */
+    @NotThreadSafe
+    static private class ThroughtputWriter {
+        final String action;
+        final int messageSize;
+        final long startTime;
+        final List<Long> throughputs;
+        long count;
+        @GuardedBy("LOCk")
+        long seconds;
+        @GuardedBy("LOCk")
+        long totalBytesInSecond;
+        long totalBytes;
+
+        final ScheduledExecutorService scheduler;
+
+        private final Object LOCk = new Object();
+
+        ThroughtputWriter(String action, int messageSize, long startTime) {
+            this.action = action;
+            this.messageSize = messageSize;
+            this.startTime = startTime;
+            this.throughputs = new ArrayList<>();
+            this.count = 0;
+
+            scheduler = Executors.newSingleThreadScheduledExecutor();
+        }
+
+        public void start() {
+            scheduler.scheduleAtFixedRate(this::recordSecondThroughput, 1, 1, TimeUnit.SECONDS);
+        }
+
+        public void stop() {
+            scheduler.shutdownNow();
+        }
+
+        public void recordSecondThroughput() {
+            synchronized (LOCk) {
+                recordThroughput(seconds, totalBytesInSecond);
+
+                totalBytesInSecond = 0;
+                seconds++;
+            }
+        }
+
+        public void recordThroughput(long seconds, long throughputBytes) {
+            throughputs.add(totalBytesInSecond);
+        }
+
+        private String[] getPercentiles() {
```

Comment:
I don't see the benefit of showing the percentiles of the throughput. There is only 1 data point per second so percentiles are overkill and confusing. I recommend just showing the throughput during the last 1 second every 1 second. If somebody wants to see the throughput during previous intervals, they can just look at previous lines.
To understand the distinction with latency, taking the mean of latencies is dangerous because it loses the most important information (slow operations). So we use percentiles to aggregate the 1000's of latencies per second.
In contract, the throughput that this is calculating is actually the mean throughput for each approximately 1 second interval. Using the mean doesn't lose important information. It is also easy enough to visually inspect the last 10 intervals to see any trends. Using percentiles to aggregate this over all previous time intervals isn't very useful.

Reply:
I've now updated the PR by removing the Scheduled Executor and the Throughput Percentiles.

The CSV throughput writer now works by getting a report at the end of every TimeWindow (the interval period is now configurable, previously it was fixed at 5 seconds).  This allows different polling frequencies to be specified for the desired throughput resolution.

Reply:
I've also removed the `,` from the Latency output as it turned out some teams were scraping these values from the logs.